grakn-core/src/main/java/ai/grakn/util/CommonUtil.java

```diff
@@ -70,6 +73,20 @@ public static void withImplicitConceptsVisible(GraknGraph graph, Runnable functi
     }
 
     /**
+     * Helper method which converts the implicit type label back into the original label from which is was built.
+     *
+     * @param implicitType the implicit type label
+     * @return The original label which was used to build this type
+     */
+    public static Label explicitLabel(Label implicitType){
+        if(!implicitType.getValue().startsWith("key") && implicitType.getValue().startsWith("has")){
+            throw new IllegalArgumentException(INVALID_IMPLICIT_TYPE.getMessage(implicitType));
+        }
+        int endIndex = implicitType.getValue().lastIndexOf("-");
```

Comment:
Just realised why this didn't break. It was designed for converting the implicit role labels only.